base/pmap.jl

```diff
@@ -0,0 +1,97 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+
+"""
+    pgenerate([::WorkerPool], f, c...) -> iterator
+
+Apply `f` to each element of `c` in parallel using available workers and tasks.
+
+For multiple collection arguments, apply f elementwise.
+
+Results are returned in order as they become available.
+
+Note that `f` must be made available to all worker processes; see
+[Code Availability and Loading Packages](:ref:`Code Availability
+and Loading Packages <man-parallel-computing-code-availability>`)
+for details.
+"""
+function pgenerate(p::WorkerPool, f, c)
+    if length(p) == 0
+        return asyncgenerate(f, c)
+    end
+    batches = batchsplit(c, min_batch_count = length(p) * 3)
+    return flatten(asyncgenerate(remote(p, b -> asyncmap(f, b)), batches))
+end
+
+pgenerate(p::WorkerPool, f, c1, c...) = pgenerate(p, a->f(a...), zip(c1, c...))
+
+pgenerate(f, c) = pgenerate(default_worker_pool(), f, c...)
+pgenerate(f, c1, c...) = pgenerate(a->f(a...), zip(c1, c...))
+
+
+
+"""
+    pmap([::WorkerPool], f, c...) -> collection
+
+Transform collection `c` by applying `f` to each element using available
+workers and tasks.
+
+For multiple collection arguments, apply f elementwise.
+"""
+pmap(p::WorkerPool, f, c...) = collect(pgenerate(p, f, c...))
+pmap(f, c...) = pmap(p, f, c...)
```

Comment:
As it stands you could do [`asyncmap(f, c)`](https://github.com/JuliaLang/julia/pull/15409/files/3568a1149486b5b4d3a9ddf31fe98bcbadbd5305#diff-b4c399b423f02e7569b7de53c31795a0R170) or `pmap(WorkerPool(), f, c)` to avoid any remote calls. I guess adding a `tasks_only=true` kw arg would be simple enough. My personal preference would be to just call `asyncmap` in that case.